forte/register_forte_options.py

```diff
@@ -214,6 +217,53 @@ def register_embedding_options(options):
     options.add_bool(
         "PAO_FIX_VIRTUAL_NUMBER", False,
         "Enable this option will generate PAOs equivlent to ASET virtuals, instead of using threshold")
+    options.add_str(
+        "EMBEDDING_TYPE", "ASET_MF", ["ASET_MF", "ASET2", "ASET-SWAP"],
+        "The type of embedding computations.")
+    options.add_str(
+        'FRAGMENT_DENSITY', 'RHF', ['CASCI', 'CASSCF', 'RHF', 'FULL'],
+        'The real/approximate RDMs used in the correlative environment computation')
+    options.add_int(
+        "ADD_FRAGMENT_DOCC", 0,
+        "Manually ddjust number of restricted occupied orbitals A_C in A for mo_space_info_A")
+    options.add_int(
+        "ADD_FRAGMENT_ACTIVE", 0,
+        "Manually ddjust number of active orbitals A_A in A for mo_space_info_A")
+    options.add_str(
+        'FRAG_CORRELATION_SOLVER', 'MRDSRG',
+        ['DSRG-MRPT2', 'THREE-DSRG-MRPT2', 'DSRG-MRPT3', 'MRDSRG',
+         'SOMRDSRG', 'MRDSRG_SO', 'DSRG_MRPT'],
+        'Dynamical correlation solver type for Fragment inactive orbitals.')
+    options.add_str(
+        'ENV_CORRELATION_SOLVER', 'MRDSRG',
+        ['DSRG-MRPT2', 'THREE-DSRG-MRPT2', 'DSRG-MRPT3', 'MRDSRG',
+         'SOMRDSRG', 'MRDSRG_SO', 'DSRG_MRPT'],
+        'Dynamical correlation solver type for Fragment-environment correlation')
+    options.add_str("FRAG_CORR_LEVEL", "LDSRG2",
+        ["PT2", "PT3", "LDSRG2", "LDSRG2_QC", "LSRG2", "SRG_PT2", "QDSRG2",
+         "LDSRG2_P3", "QDSRG2_P3"],
+        "Correlation level of fragment MR-DSRG (used in mrdsrg code, "
+        "LDSRG2_P3 and QDSRG2_P3 not implemented)")
+    options.add_str("ENV_CORR_LEVEL", "PT2",
+        ["PT2", "PT3", "LDSRG2", "LDSRG2_QC", "LSRG2", "SRG_PT2", "QDSRG2",
+         "LDSRG2_P3", "QDSRG2_P3"],
+        "Correlation level of environment (interactive) MR-DSRG (used in mrdsrg code, "
+        "LDSRG2_P3 and QDSRG2_P3 not implemented)")
+    options.add_bool(
+        "embedding_disable_semi_check", True,
+        "Whether we skip semi-canonicalization check in DSRG code when computing ASET(2).")
+    options.add_bool(
+        "embedding_align_scalar", False,
+        "Do write shifts to align the integral scalar.")
```

Comment:
What is an integral scalar? What does it mean to align a scalar? Where are the shifts written to? What is a shift, and how does it affect alignment of a scalar?

In short, I have no idea what this option is supposed to do.

Reply:
The scalar is simply frozen core energy + nuclear repulsion energy. Those values should be added when doing "fragment (A) only" computations, while when we do "A + B", this scalar should be off, since they are already computed when building integrals.
So this is another "internal options" only used to communicate within my codes, not for the user.